### PR TITLE
Fix_lab3

### PR DESCRIPTION
**Pull request by Nandi1218**
3 bugs fixed in TorpedoStore.java

### CODE REVIEW

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 1]
```diff
@@ -30,7 +30,7 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos");
```
Comment (NBiTech) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:34]: Impressive work!

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 2]
```diff
@@ -8,9 +8,10 @@
 * (Deliberately contains bugs.)
 */
 public class TorpedoStore {
-
+  
   // rate of failing to fire torpedos [0.0, 1.0]
   private double FAILURE_RATE = 0.0; //NOSONAR
+  private Random generator = new Random();
```
Comment (NBiTech) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:14]: Excellent!

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 3]
```diff
@@ -41,7 +41,7 @@ public boolean fire(int numberOfTorpedos){
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
```
Comment (NBiTech) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:44]: Nicely done

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 4]
```diff
@@ -8,9 +8,10 @@
 * (Deliberately contains bugs.)
 */
 public class TorpedoStore {
-
+
```
Comment (NBiTech) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:11]: Approve

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 5]
```diff
@@ -30,7 +30,7 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
```
Comment (NBiTech) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:33]: Ok

**Review by NBiTech — CHANGES_REQUESTED** [review]
Nicely done, hope you didn't clock in for this :)

**Review by NBiTech — APPROVED** [review]
All good here.

File: src/main/java/hu/bme/mit/spaceship/TorpedoStore.java [hunk 6]
```diff
@@ -30,18 +31,17 @@ public TorpedoStore(int numberOfTorpedos){
...
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount -= numberOfTorpedos;
       success = true;
     } else {
       // simulated failure
```
Comment (hudakmariann) [inline src/main/java/hu/bme/mit/spaceship/TorpedoStore.java:47]: // Random komment

**Review by hudakmariann — COMMENTED** [review]
Accepted!

**Pull request merged**